Feature/product page

bojkovladislav opened the pull request:
# Feature: Responsive Product Page

This pull request implements a fully **responsive Product Page** for the application, ensuring a seamless experience across mobile, tablet, and desktop devices.

bojkovladislav committed:
fix: add links for product card

bojkovladislav committed:
fix: add links for product card

bojkovladislav committed:
feat: Product feature introduced

bojkovladislav committed:
feat: Product feature introduced

bojkovladislav committed:
fix: update paths

bojkovladislav committed:
fix: update paths

bojkovladislav committed:
feat: create carousel feature

bojkovladislav committed:
feat: create carousel feature

bojkovladislav committed:
fix: center LeftArrowIcon

bojkovladislav committed:
fix: center LeftArrowIcon

bojkovladislav committed:
chore: create styling file for the product page

bojkovladislav committed:
chore: create styling file for the product page

bojkovladislav committed:
feat: add carousel

bojkovladislav committed:
feat: add carousel

bojkovladislav committed:
fix: reuse layout & business logic

bojkovladislav committed:
fix: reuse layout & business logic

bojkovladislav committed:
chore: feat: add mobile version of product page

bojkovladislav committed:
chore: feat: add mobile version of product page

bojkovladislav committed:
feat: add mobile version of reviews

bojkovladislav committed:
feat: add mobile version of reviews

bojkovladislav committed:
fix: products mockData & add reviews

bojkovladislav committed:
fix: products mockData & add reviews

bojkovladislav committed:
feat: add new reusable components

bojkovladislav committed:
feat: add new reusable components

bojkovladislav committed:
feat: add review type

bojkovladislav committed:
feat: add review type

bojkovladislav committed:
chore: add new icons

bojkovladislav committed:
chore: add new icons

bojkovladislav committed:
fix: minor fixes

bojkovladislav committed:
fix: minor fixes

bojkovladislav committed:
fix: arrow alignment

bojkovladislav committed:
fix: add responsiveness to carousel

bojkovladislav committed:
add responsiveness to the product page

bojkovladislav committed:
fix: move Breadcrumbs to layout & add responsiveness

bojkovladislav committed:
fix: add responsiveness to reviews

bojkovladislav committed:
feat: introduce new ProductImageGalley feature

bojkovladislav committed:
fix: minor fixes

Virass committed:
fix: resolve merge conflicts in favor of remote (theirs)

Virass commented on src/features/website/ProductImageGallery/ProductImageGallery.tsx:
```diff
@@ -0,0 +1,52 @@
+
+    const selectImage = (img: string) => {
+        setSelectedImage(img);
+    };
```
Це просто функція яка викликає функцію.

Virass commented on src/features/website/Product/Carousel/Carousel.tsx:
Якщо ця функціональність відноситься до фічі, то вона має знаходитися в папці самої фічі. Якщо це перевикористовуваний компонент то в src/shared/components

Virass commented on src/features/website/Product/AvailableColors/AvailableColors.tsx:
```diff
@@ -0,0 +1,56 @@
+
+    const selectColor = (color: string) => {
+        setSelectedColor(color);
+    };
```
Функція яка просто викликає іншу функцію

Virass committed:
refactor: remove Carousel component and related styles

- Deleted the Carousel component and its associated styles from the project.
- Updated the AvailableColors component to simplify color selection logic.
- Adjusted imports in the Product component to reflect the Carousel removal.
- Added a placeholder image constant and SVG for better image handling in the ProductImageGallery.
